tests/app/questionnaire/test_path_finder.py

```diff
@@ -653,6 +653,124 @@ def test_repeating_groups_conditional_location_path(self):
 
             self.assertEqual(expected_path, path_finder.get_full_routing_path())
 
+    def test_repeating_group_block_skip_correct_if_independent_answer_is_not_repeating_but_causes_a_skip(self):
+        """ tests blocks are correctly added to path in the case where the blocks in a repeating group have
+        skip conditions and those skip conditions are dependant on a non repeating answer which causes a skip
+        condition to return true.
+        """
+
+        independent_answer = '3'
+
+        with patch('app.questionnaire.rules._answer_is_in_repeating_group', return_value=False):
+
+            expected_path = [
+                Location('about-household-group', 0, 'household-composition'),
+                Location('independent-answer-group', 0, 'non-repeating-question-block'),
+                Location('questionnaire-completed', 0, 'confirmation')
+            ]
+
+            schema = load_schema_from_params('test', 'skip_conditions_from_repeating_group_based_on_non_repeating_answer')
+            answer_store = AnswerStore()
+
+            # Set up some first names
+            answer_store.add(Answer(answer_id='first-name', value='aaa', group_instance=0))
+            answer_store.add(Answer(answer_id='first-name', value='bbb', group_instance=1))
+            answer_store.add(Answer(answer_id='first-name', value='ccc', group_instance=2))
+
+            # Set up the independent answer
+            answer_store.add(Answer(answer_id='an-independent-answer', value=independent_answer, group_instance=0))
+
+            sut = PathFinder(schema, answer_store=answer_store, metadata={}, completed_blocks=[])
```

Comment:
There was a bunch that in secure-messaging and I didn't really understand it there either.  It's easier to actually call the thing what it's called instead of using the generic term for it

Reply:
The rest of that test file just does `path_finder = PathFinder(...`, agree with @insacuri